File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
from sys import stdin
dict = {}
n = int(stdin.readline())

for _ in range(n):
    # 직원 이름과 enter / leave 여부 딕셔너리에 저장
    a,b = stdin.readline().split()
    dict[a] = b

result = []

for i in dict:
    # 딕셔너리 value == enter인 것만 리스트에 삽입(아직 퇴근 안했다는 뜻)
    if dict[i] == 'enter':
        result.append(i)

# 역순으로 출력
result.sort(reverse=True)

for j in result:
    print(j)

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int N = Integer.parseInt(br.readLine());

        HashMap<String, Integer> map = new HashMap<String, Integer>();

        String name;
        String status;

        for(int i = 0; i < N; i++){
            st = new StringTokenizer(br.readLine());
            name = st.nextToken();
            status = st.nextToken();
            if(map.containsKey(name) && status.equals("leave")){ // 퇴근하는 경우
                map.remove(name);
            }
            else if (!map.containsKey(name) && status.equals("enter")){ // 출근하는 경우
                map.put(name, 1);
            }
        }

        br.close();

        String[] strArray = map.keySet().toArray(new String[map.size()]);

        Arrays.sort(strArray, Collections.reverseOrder());

        for(String active: strArray){
            System.out.println(active);
        }
    }
}